code/go/0chain.net/smartcontract/dbs/event/allocation_test.go

```diff
@@ -0,0 +1,398 @@
+		// allocation should be locked for the blobber rewards even if
+		// user never write something to the blobber.
+		MinLockDemand float64 `json:"min_lock_demand"`
+		// MaxOfferDuration with this prices and the demand.
+		MaxOfferDuration time.Duration `json:"max_offer_duration"`
+		// ChallengeCompletionTime is duration required to complete a challenge.
+		ChallengeCompletionTime time.Duration `json:"challenge_completion_time"`
+	}
+
+	type PriceRange struct {
+		Min state.Balance `json:"min"`
+		Max state.Balance `json:"max"`
+	}
+
+	type StorageNode struct {
+		ID              string                 `json:"id"`
+		BaseURL         string                 `json:"url"`
+		Geolocation     StorageNodeGeolocation `json:"geolocation"`
+		Terms           Terms                  `json:"terms"`    // terms
+		Capacity        int64                  `json:"capacity"` // total blobber capacity
+		Used            int64                  `json:"used"`     // allocated capacity
+		LastHealthCheck common.Timestamp       `json:"last_health_check"`
+		PublicKey       string                 `json:"-"`
+		// StakePoolSettings used initially to create and setup stake pool.
+		StakePoolSettings stakePoolSettings `json:"stake_pool_settings"`
+	}
+
+	type StorageAllocationStats struct {
+		UsedSize                  int64  `json:"used_size"`
+		NumWrites                 int64  `json:"num_of_writes"`
+		NumReads                  int64  `json:"num_of_reads"`
+		TotalChallenges           int64  `json:"total_challenges"`
+		OpenChallenges            int64  `json:"num_open_challenges"`
+		SuccessChallenges         int64  `json:"num_success_challenges"`
+		FailedChallenges          int64  `json:"num_failed_challenges"`
+		LastestClosedChallengeTxn string `json:"latest_closed_challenge"`
+	}
+
+	type WriteMarker struct {
+		AllocationRoot         string           `json:"allocation_root"`
+		PreviousAllocationRoot string           `json:"prev_allocation_root"`
+		AllocationID           string           `json:"allocation_id"`
+		Size                   int64            `json:"size"`
+		BlobberID              string           `json:"blobber_id"`
+		Timestamp              common.Timestamp `json:"timestamp"`
+		ClientID               string           `json:"client_id"`
+		Signature              string           `json:"signature"`
+	}
+
+	type BlobberAllocation struct {
+		BlobberID       string                  `json:"blobber_id"`
+		AllocationID    string                  `json:"allocation_id"`
+		Size            int64                   `json:"size"`
+		AllocationRoot  string                  `json:"allocation_root"`
+		LastWriteMarker *WriteMarker            `json:"write_marker"`
+		Stats           *StorageAllocationStats `json:"stats"`
+		Terms           Terms                   `json:"terms"`
+		// MinLockDemand for the allocation in tokens.
+		MinLockDemand state.Balance `json:"min_lock_demand"`
+		Spent         state.Balance `json:"spent"`
```

Comment:
Do we need to define them all again here in tests?

Reply:
cyclic dependency

Reply:
@Sriep @ssardana08 Should we plan to abstract out the models into a different package to avoid re-declaration and remove the redundancy from code? This way we need to maintain two models out of which one is for ensuring if tests are working which is against the DRY.

Reply:
The is this issue https://github.com/0chain/0chain/issues/522.